presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowAbstractMetadata.java

```diff
@@ -0,0 +1,321 @@
+import com.facebook.presto.common.type.VarbinaryType;
+import com.facebook.presto.common.type.VarcharType;
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorTableHandle;
+import com.facebook.presto.spi.ConnectorTableLayout;
+import com.facebook.presto.spi.ConnectorTableLayoutHandle;
+import com.facebook.presto.spi.ConnectorTableLayoutResult;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.Constraint;
+import com.facebook.presto.spi.NotFoundException;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.SchemaTablePrefix;
+import com.facebook.presto.spi.connector.ConnectorMetadata;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.apache.arrow.flight.FlightInfo;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.plugin.arrow.ArrowErrorCode.ARROW_FLIGHT_ERROR;
+import static java.util.Objects.requireNonNull;
+
+public abstract class ArrowAbstractMetadata
+        implements ConnectorMetadata
+{
+    private static final Logger logger = Logger.get(ArrowAbstractMetadata.class);
+    private final ArrowFlightConfig config;
+    private final ArrowFlightClientHandler clientHandler;
+
+    public ArrowAbstractMetadata(ArrowFlightConfig config, ArrowFlightClientHandler clientHandler)
+    {
+        this.config = config;
+        this.clientHandler = requireNonNull(clientHandler);
+    }
+
+    protected abstract String getDBSpecificSchemaName(ArrowFlightConfig config, String schemaName);
+
+    protected abstract String getDBSpecificTableName(ArrowFlightConfig config, String tableName);
+
+    @Override
+    public ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName)
+    {
+        if (!listSchemaNames(session).contains(tableName.getSchemaName())) {
+            return null;
+        }
+
+        if (!listTables(session, Optional.ofNullable(tableName.getSchemaName())).contains(tableName)) {
+            return null;
+        }
+        return new ArrowTableHandle(tableName.getSchemaName(), tableName.getTableName());
+    }
+
+    protected abstract ArrowFlightRequest getArrowFlightRequest(ArrowFlightConfig config, Optional<String> query, String schema, String table);
+
+    protected abstract ArrowFlightRequest getArrowFlightRequest(ArrowFlightConfig config, String schema);
+
+    public List<Field> getColumnsList(String schema, String table, ConnectorSession connectorSession)
+    {
+        try {
+            String dbSpecificSchemaName = getDBSpecificSchemaName(config, schema);
```

Comment:
Avoid "DB" references as it might not be an underlying DB

Reply:
I have removed "DB" references